packages/overlay/src/vaadin-overlay-mixin.js

```diff
@@ -568,7 +568,7 @@ export const OverlayMixin = (superClass) =>
         });
         this.dispatchEvent(evt);
 
-        if (this.opened && !evt.defaultPrevented) {
+        if (this.opened && !evt.defaultPrevented && !event.defaultPrevented) {
```

Comment:
I wonder if we should dispatch `vaadin-overlay-escape-press` at all in this case since as it won't take any effect regardless? Especially because this event isn't forwarded to the body, unlike some other events.